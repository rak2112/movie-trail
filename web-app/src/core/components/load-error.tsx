import React from 'react';
import { Api } from '../interfaces';


export const LoadError = ({hasError, errorMessage}: Api) => (
  <>
  {
    hasError && 
    <div style={{margin:0, fontSize: '1.8rem', textAlign: 'center'}} className="alert alert-danger" role="alert">
      {errorMessage}
    </div>
  }
  </>
);