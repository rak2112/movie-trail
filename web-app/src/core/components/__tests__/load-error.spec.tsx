
import React from 'react';
import { create } from 'react-test-renderer';
import { LoadError } from '../load-error';

describe(`LoadError Component`, () => {
  const defaultProps = {
    hasError: true,
    errorMessage: 'Error, something has gone wrong'
  };

  it(`should render with given inputs`, () => {
    const wrapper = create(
      <LoadError {...defaultProps}/>
    ).toJSON();
    
    expect(wrapper).toMatchSnapshot();
  });

  it(`should render with given inputs`, () => {
    const props = { ...defaultProps, hasError: false }; 
    const wrapper = create(
      <LoadError {...props}/>
    ).toJSON();
    
    expect(wrapper).toMatchSnapshot();
  });
});