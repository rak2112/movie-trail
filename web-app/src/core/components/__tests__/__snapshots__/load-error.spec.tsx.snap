// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadError Component should render with given inputs 1`] = `
<div
  className="alert alert-danger"
  role="alert"
  style={
    Object {
      "fontSize": "1.8rem",
      "margin": 0,
      "textAlign": "center",
    }
  }
>
  Error, something has gone wrong
</div>
`;

exports[`LoadError Component should render with given inputs 2`] = `null`;
